### PR TITLE
logic error

### PR DESCRIPTION
**Pull request by foxyriver**
**What this PR does / why we need it**:

if err != nil , we want return err. but it can't achieve this in if branch, should achieve this in else branch.

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: pkg/genericapiserver/api/handlers/rest.go [hunk 1]
```diff
@@ -1044,6 +1041,8 @@ func checkName(obj runtime.Object, name, namespace string, namer ScopeNamer) err
 					"the namespace of the object (%s) does not match the namespace on the request (%s)", objNamespace, namespace))
 			}
 		}
+	} else {
+		return err
```
Comment (lavalamp) [inline pkg/genericapiserver/api/handlers/rest.go]: The more usual go style is to put the error case first, e.g.

```go
if a, b, c, err := f(); err != nil {
  return err
} else {
  // use a, b, c
}
```

**Comment by lavalamp** [comment]
@k8s-bot ok to test

**Comment by lavalamp** [comment]
sorry for the big delay, I have been OOO.

File: pkg/genericapiserver/api/handlers/rest.go (hunk 1 above)
Comment (foxyriver) [inline pkg/genericapiserver/api/handlers/rest.go]: thx, I slipped up there, I will modify it.

**Comment by foxyriver** [comment]
@lavalamp thx for your answer, i have changed it, please recheck.

**Comment by lavalamp** [comment]
@k8s-bot ok to test

/lgtm

**Comment by foxyriver** [comment]
@lavalamp rebase again, PTAL

**Comment by lavalamp** [comment]
/lgtm

@k8s-bot bazel test this

**Comment by sttts** [comment]
Bazel is unhappy.

**Comment by foxyriver** [comment]
@lavalamp PTAL

**Comment by sttts** [comment]
@k8s-bot unit test this

**Comment by sttts** [comment]
@k8s-bot unit test this

**Comment by lavalamp** [comment]
@k8s-bot unit test this

File: staging/src/k8s.io/apiserver/pkg/endpoints/handlers/rest.go [hunk 2]
```diff
@@ -1076,10 +1076,9 @@ func hasUID(obj runtime.Object) (bool, error) {
 
 // checkName checks the provided name against the request
 func checkName(obj runtime.Object, name, namespace string, namer ScopeNamer) error {
-	if objNamespace, objName, err := namer.ObjectName(obj); err == nil {
```
Comment (kargakis) [inline staging/src/k8s.io/apiserver/pkg/endpoints/handlers/rest.go]: The following code seems simpler
```go
objNamespace, objName, err := namer.ObjectName(obj)
if err != nil {
    return err
}
if objName != name {
	return errors.NewBadRequest(fmt.Sprintf("the name of the object (%s) does not match the name on the URL (%s)", objName, name))
}
if len(namespace)>0 && len(objNamespace) > 0 && objNamespace != namespace {
    return errors.NewBadRequest(fmt.Sprintf("the namespace of the object (%s) does not match the namespace on the request (%s)", objNamespace, namespace))
}
```

Comment (kargakis) [inline staging/src/k8s.io/apiserver/pkg/endpoints/handlers/rest.go]: Also while here is namer.ObjectName returning api errors?

**Pull request closed without merging**